src/settings/environment.rs

```diff
@@ -0,0 +1,109 @@
+use std::collections::HashMap;
+use std::env;
+
+use config::{ConfigError, Source, Value};
+
+pub trait QueryEnvironment {
+    fn get_var(&self, var: &'static str) -> Result<String, std::env::VarError>;
+
+    fn empty(&self) -> Result<bool, failure::Error>;
+}
+
+#[derive(Clone, Debug)]
+pub struct Environment {
+    whitelist: Vec<&'static str>,
+}
+
+impl Environment {
+    pub fn with_whitelist(whitelist: Vec<&'static str>) -> Self {
+        Environment { whitelist }
+    }
+}
+
+impl QueryEnvironment for Environment {
+    fn get_var(&self, var: &'static str) -> Result<String, std::env::VarError> {
+        env::var(var)
+    }
+
+    fn empty(&self) -> Result<bool, failure::Error> {
+        let env = self.collect()?;
+
+        Ok(env.is_empty())
+    }
+}
+
+// Source trait implementation for use with Config::merge
+// until config crate removal is complete. This is effectively a
+// copy of the config crate's impl of Source for its Environment
+// struct, but rather than pulling the whole environment in #collect,
+// we pull in only whitelisted values, and rather than taking a custom
+// prefix, we assume a prefix of `CF_`.
+impl Source for Environment {
```

Comment:
keeping the `collect` implementation is nice, though, so perhaps this will not go away in the long term, only behave differently.